addon/store.js

```diff
@@ -61,6 +61,15 @@ var Store = ServiceType.extend({
       });
       this.set("array", {});
     },
+    willDestroy() {
+      this.setProperties({
+        "array": null,
+        "identityMap": null,
+        // Would rather set to null but updateFilters runs after destroy somehow
```

Comment:
I'm open to the isDestroying/destroyed checks you mention (as I assume this helps avoid the run.next requirement you show)